tests/test_bin_trie.py

```diff
@@ -0,0 +1,51 @@
+import random
+
+from hypothesis import (
+    given,
+    strategies as st,
+    settings,
+)
+
+from eth_utils import (
+    to_list,
+)
+
+from trie.trie import (
+    BinaryTrie,
+)
+from trie.constants import (
+    BLANK_HASH,
+)
+
+@to_list
+def to_kv_pairs(kv_zip):
+    for k, v in kv_zip:
+        yield (k, v)
+
+
+@given(k=st.lists(st.binary(min_size=32, max_size=32), min_size=100, max_size=100, unique=True),
+    v=st.lists(st.binary(min_size=1), min_size=100, max_size=100),
+    set_random=st.random_module())
+@settings(max_examples=10)
+def test_bin_trie_different_order_insert(k, v, set_random):
+    kv_pairs = to_kv_pairs(zip(k, v))
+    result = BLANK_HASH
+    # Repeat 3 times
+    for _ in range(3):
+        trie = BinaryTrie(db={})
+        random.shuffle(kv_pairs)
```

Comment:
After look into hypothesis documents, it seems that `strategy.random_module` seed the `random` module at the start of the test and what it returns is an object that basically represents the seed (and hence no `shuffle` function) while `strategy.randoms` returns a random number generator. I will change it to `strategy.randoms` since the code does not depend on the global`random` module.

Reply:
So it is.  Sorry to have mislead you.  Carry on!